### PR TITLE
refactor!: Remove deprecated parameters constructors (Rework charge types, Part 2)

### PR DESCRIPTION
**Pull request by msmk0**
Remove deprecated constructors from track parameters types and update the constructor calls mostly in the tests.

This is a follow up to #437 and needs to be merged afterwards.

### CODE REVIEW

**Comment by msmk0** [comment]
@asalzburger The builds go through but there are a few unit tests failure. I do not really understand where they are coming from.

**Comment by msmk0** [comment]
@asalzburger  The unit test failures are fixed. I had to disable one failure in the `ImpactPointEstimator3D` test that I do not understand; this is tracked in #455 and needs to be followed up by @baschlag afterwards.

**Comment by asalzburger** [comment]
Alright, I will review this morning.

File: Tests/UnitTests/Core/Propagator/AtlasStepperTests.cpp [hunk 1]
```diff
@@ -52,11 +53,11 @@ static constexpr NavigationDirection navDir = backward;
 static const ConstantBField magneticField(Vector3D(0.1_T, -0.2_T, 2_T));
 
 // initial parameter state
-static const Vector3D pos(1_mm, -1_mm, 2_mm);
-static constexpr auto time = 2_ns;
+static const Vector4D pos4(1_mm, -1_mm, 2_mm, 2_ns);
+static const Vector3D pos = pos4.segment<3>(ePos0);
```
Comment (asalzburger) [inline Tests/UnitTests/Core/Propagator/AtlasStepperTests.cpp:57]: Just for my understanding, what's the difference between `.segment<int DIM>` and `.block<int ROWS, int COLS>` - is the second just restricted for 1-row and or 1-col types?

**Review by asalzburger — APPROVED** [review]
Very nice and clean PR - I really like the new constructors, I found it always quite strange to create `Parameters(std::nullopt, ...pack)`.

Happily approving.

File: Tests/UnitTests/Core/Propagator/AtlasStepperTests.cpp (hunk 1 above)
Comment (msmk0) [reply Tests/UnitTests/Core/Propagator/AtlasStepperTests.cpp:57]: `.segment<...>()` only exists for vectors, i.e. 1d matrices while `.block<..., ...>()` can always be called.

**Comment by asalzburger** [comment]
I will update the branch with the KF changes and wait for the first build to run through.

File: Tests/UnitTests/Core/Propagator/AtlasStepperTests.cpp (hunk 1 above)
Comment (asalzburger) [reply Tests/UnitTests/Core/Propagator/AtlasStepperTests.cpp:57]: Thanks. As it requires one less template parameter - might that be slightly better in term of compilation resources?

**Comment by msmk0** [comment]
@asalzburger Ok, when you merge it could you use the PR description instead of the list of commits as the merge message?

File: Tests/UnitTests/Core/Propagator/AtlasStepperTests.cpp (hunk 1 above)
Comment (msmk0) [reply Tests/UnitTests/Core/Propagator/AtlasStepperTests.cpp:57]: I would be surprised if that is the case. The underlying matrix type still has two dimension parameters; one is just 1.

**Pull request merged**